ui/plugin-system/src/runtime/datasources.ts

```diff
@@ -27,10 +27,16 @@ export interface DatasourceStore {
   /**
    * Gets a list of datasource metadata for a plugin kind.
    */
-  listDatasourceMetadata(datasourcePluginKind: string): Promise<DatasourceMetadata[]>;
+  listDatasourceMetadata(datasourcePluginKind: string): Promise<DatasourceMetadata>;
 }
 
-export interface DatasourceMetadata {
+export type DatasourceMetadata = Array<{
```

Comment:
Rename ``DatasourceMetadata`` and ``DatasourceMetadataEntry`` to something more meaningful like 

``DatasourceSelectItemGroup[]`` and ``DatasourceSelectItem[]``